lib/protein/amqp_adapter.rb

```diff
@@ -34,6 +32,10 @@ def timeout(timeout = :not_set)
       instance_variable_defined?("@timeout") ? @timeout : 15_000
     end
 
+    def init
```

Comment:
Looks kinda funny considering the built-in `initialize` - kinda like a typo from someone who wrote too much Elixir lately :)

(...but it's fine of course)

Reply:
Since this is needed to perform some setup on startup I'd prefer to rewrite both `AMQPAdapter` and `HTTPAdapter` to have configurable instances, but this would have implications of breaking public interface -- when the test suite is more complete it will be much safer to do such changes. For now I decided to stick to a bit faulty idea of custom `init` function.

Reply:
Well, I didn't want to rewrite the library just to make it clearer. The problem is, that `AMQPAdapter` is a singleton in terms of every class that inherits from it.

I'd prefer to instantiate transport classes explicitly with injected configuration (then there would be natural place to initialize anything, like Amqp connection mutex), but this refactoring is completely outside of the scope of what I'm fixing.